tree/dataframe/test/dataframe_merge_results.cxx

```diff
@@ -0,0 +1,172 @@
+
+    mergeablecount->Merge(othermergeablecount);
+    auto mergedcount = mergeablecount->GetValue();
+
+    EXPECT_EQ(mergedcount, 200);
+}
+
+TEST(RDataFrameMergeResults, MergeMean)
+{
+    ROOT::RDataFrame df1{10};
+    ROOT::RDataFrame df2{20};
+    
+    auto col1 = df1.Define("x","double(rdfentry_)");
+    auto col2 = df2.Define("x","double(rdfentry_)");
+
+    auto mean1 = col1.Mean("x");
+    auto mean2 = col2.Mean("x");
+
+    auto mergeablemean = ROOT::Detail::RDF::GetMergeableValue(mean1);
+    auto othermergeablemean = ROOT::Detail::RDF::GetMergeableValue(mean2);
+
+    mergeablemean->Merge(othermergeablemean);
+
+    auto mergedmean = mergeablemean->GetValue();
+
+    EXPECT_NEAR(mergedmean, 7.83333333333, 0.01);
+}
+
+TEST(RDataFrameMergeResults, MergeHisto1DSimple)
+{
+    ROOT::RDataFrame df1{100};
+    ROOT::RDataFrame df2{100};
+    
+    auto col1 = df1.Define("x","double(rdfentry_)");
+    auto col2 = df2.Define("x","double(rdfentry_)");
+
+    auto hist1 = col1.Histo1D("x");
+    auto hist2 = col2.Histo1D("x");
+
+    auto mergeablehist = ROOT::Detail::RDF::GetMergeableValue(hist1);
+    auto othermergeablehist = ROOT::Detail::RDF::GetMergeableValue(hist2);
+
+    mergeablehist->Merge(othermergeablehist);
+
+    auto &mergedhist = mergeablehist->GetValue();
+
+    EXPECT_EQ(mergedhist.GetEntries(), 200);
+    EXPECT_NEAR(mergedhist.GetMean(), 50, 0.5);
+}
+
+TEST(RDataFrameMergeResults, MergeHisto1DModel)
+{
+    ROOT::RDataFrame df1{100};
+    ROOT::RDataFrame df2{100};
+    
+    auto col1 = df1.Define("x","double(rdfentry_)");
+    auto col2 = df2.Define("x","double(rdfentry_)");
+
+    auto hist1 = col1.Histo1D({"name","title",100,0,100}, "x");
+    auto hist2 = col2.Histo1D({"name","title",100,0,100}, "x");
+
+    auto mergeablehist = ROOT::Detail::RDF::GetMergeableValue(hist1);
+    auto othermergeablehist = ROOT::Detail::RDF::GetMergeableValue(hist2);
+
+    mergeablehist->Merge(othermergeablehist);
+
+    auto &mergedhist = mergeablehist->GetValue();
+
+    EXPECT_EQ(mergedhist.GetEntries(), 200);
+    EXPECT_NEAR(mergedhist.GetMean(), 50, 0.5);
+}
+
+TEST(RDataFrameMergeResults, MergeHisto2D)
+{
+    ROOT::RDataFrame df{100};
+
+    auto col1 = df.Define("x","double(rdfentry_)");
+    auto col2 = col1.Define("y","double(rdfentry_)");
+
+    auto hist1 = col2.Histo2D({"name","title",100,0,100,100,0,100},"x","y");
+    auto hist2 = col2.Histo2D({"name","title",100,0,100,100,0,100},"x","y");
+
+
+    auto mergeablehist =  ROOT::Detail::RDF::GetMergeableValue(hist1);
+    auto othermergeablehist =  ROOT::Detail::RDF::GetMergeableValue(hist2);
+
+    mergeablehist->Merge(othermergeablehist);
+
+    const auto &mergedhist = mergeablehist->GetValue();
+
+    EXPECT_EQ(mergedhist.GetEntries(), 200);
+}
+
+TEST(RDataFrameMergeResults, MergeSum)
+{
+    ROOT::RDataFrame df{100};
+
+    auto sum1 = df.Sum("rdfentry_");
+    auto sum2 = df.Sum("rdfentry_");
```

Comment:
can use `Sum<ULong64_t>` to get the exact type of `rdfentry_` instead of a double